node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -413,7 +455,7 @@ async fn distribute_collation<Context>(
 
 	// Determine the group on that core.
 	let current_validators =
-		determine_our_validators(ctx, runtime, our_core, num_cores, relay_parent).await?;
```

Comment:
I think we have to pick validators based on the relay parent. Reasoning: With asynchronous backing we don't actually know when the candidate will be backed on chain, hence we can only reliably pick responsible validators based on the relay parent.